src/main/scala/edu/ie3/powerFactory2psdm/model/powerfactory/PowerFactoryGridMaps.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.powerFactory2psdm.model.powerfactory
+
+import edu.ie3.powerFactory2psdm.exception.pf.{
+  MissingGridElementException,
+  MissingParameterException
+}
+import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid.{
+  Lines,
+  Nodes,
+  Switches
+}
+import java.util.UUID
+import com.typesafe.scalalogging.LazyLogging
+
+class PowerFactoryGridMaps(pfGrid: PowerFactoryGrid) extends LazyLogging {
+
+  val UUID2node: Map[UUID, Nodes] = pfGrid.nodes match {
+    case Some(nodes) => nodes.map(node => (UUID.randomUUID(), node)).toMap
+    case None =>
+      throw MissingGridElementException("There are no nodes in the Grid")
+  }
+
+  val nodeId2UUID: Map[String, UUID] = UUID2node.map(
+    elem =>
```

Comment:
If you would use

```suggestion
    case (uuid, node) =>
```

your code gets more expressive.